tests/expected/loop-contract/nested_loop_local_var_assign.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// kani-flags: -Z loop-contracts
+
+//! Check if loop assign clause can be infered for inner-loop when there are local variables of outter-loop body.
+
+#![feature(proc_macro_hygiene)]
+#![feature(stmt_expr_attributes)]
+
+fn func(x: usize) {
+    let mut j = 0;
+    const CHUNK_SIZE: usize = 32;
+    #[kani::loop_invariant(j <=x )]
+    while j < x {
+        let mut i = 0;
+        let mut sum = 0_usize;
```

Comment:
To create a concrete example of my question above: would this test still pass if you changed this code to be:
````rust
let mut i = 0;
let mut sum : usize = i;
````